### PR TITLE
feat(STONEINTG-698): Stop provisioning ephemeral environments

### PR DESCRIPTION
**Pull request by dirgim**
* Completely stop provisioning Ephemeral environments as part of integration testing in the Snapshot controller and update the unit tests
* Treat IntegrationTestScenarios with environments defined same as the ones without environments
* Remove unused environment-related functions from the gitops package
* Update the Snapshot controller diagram

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

**Comment by hongweiliu17** [comment]
The current code looks good to me, but I feel we should also remove FindAvailableDeploymentTargetClass() and its unittest from [loader](https://github.com/redhat-appstudio/integration-service/blob/main/loader/loader.go#L49) package because they are used by `EnsureCreationOfEphemeralEnvironments()` only.

File: docs/snapshot-controller.md [hunk 1]
```diff
@@ -10,13 +10,11 @@ flowchart TD
...
-  %%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureStaticIntegrationPipelineRunsExist() function
+  %%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureIntegrationPipelineRunsExist() function
 
   %% Node definitions
   ensure1(Process further if: Snapshot testing <br>is not finished yet)
   are_there_any_ITS{"Are there any <br>IntegrationTestScenario <br>present for the given <br>Application?"}
-  does_ITS_has_env_defined{Does the <br>IntegrationTestScenario <br>has any environment <br>defined in it?}
-  skip_creating_test_PLR(Skip creating Test PLR for this ITS,<br> as it will be created by binding controller)
   create_new_test_PLR(<b>Create a new Test PipelineRun</b> for each <br>of the above ITS, if it doesn't exists already)
   mark_snapshot_InProgress(<b>Mark</b> Snapshot's Integration-testing <br>status as 'InProgress')
```
Comment (hongweiliu17) [inline docs/snapshot-controller.md:19]: I realize the `<b> </b>` is being used incorrectly in snapshot controller diagram https://github.com/redhat-appstudio/integration-service/blob/main/docs/snapshot-controller.md, but it should not be included in this PR. We may fix it after merging this PR.

**Comment by dirgim** [comment]
Marking as draft since we need to wait for the replacement solution in [RHTAPWATCH-740](https://issues.redhat.com/browse/RHTAPWATCH-740) to be implemented.

File: controllers/snapshot/snapshot_adapter_test.go [hunk 2]
```diff
@@ -62,10 +62,8 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
 		deploymentTargetClass                     *applicationapiv1alpha1.DeploymentTargetClass
 		integrationTestScenario                   *v1beta1.IntegrationTestScenario
 		integrationTestScenarioWithoutEnv         *v1beta1.IntegrationTestScenario
```
Comment (jsztuka) [inline controllers/snapshot/snapshot_adapter_test.go:64]: I guess, this and connected tests will be also removed in upcoming changes for 841.

File: controllers/snapshot/snapshot_adapter.go [hunk 3]
```diff
@@ -127,41 +127,16 @@ func (a *Adapter) EnsureRerunPipelineRunsExist() (controller.OperationResult, er
...
-				// remove label to prevent indefinite failures, users must restart test manually again
-				if err = gitops.RemoveIntegrationTestRerunLabel(a.client, a.context, a.snapshot); err != nil {
-					return controller.RequeueWithError(err)
-				}
-				return controller.StopProcessing()
-			}
-			return controller.RequeueWithError(fmt.Errorf("failed to ensure existence of the environment for scenario %s: %w", integrationTestScenario.Name, err))
-		}
-
-	} else {
```
Comment (jsztuka) [inline controllers/snapshot/snapshot_adapter.go:151]: Why are we removing else statement content there? I'm missing how is it connected to ephemeral env here.

**Review by jsztuka — APPROVED** [review]
lgtm

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by MartinBasti — APPROVED** [review]
